src/lib/sql/sql_pipeline.hpp

```diff
@@ -32,30 +32,9 @@ namespace opossum {
  */
 class SQLPipeline : public Noncopyable {
  public:
-  // No explicit transaction context constructors
-  explicit SQLPipeline(const std::string& sql, const UseMvcc use_mvcc = UseMvcc::Yes);
-
-  SQLPipeline(const std::string& sql, const std::shared_ptr<Optimizer>& optimizer,
-              const UseMvcc use_mvcc = UseMvcc::Yes);
-
-  SQLPipeline(const std::string& sql, const PreparedStatementCache& prepared_statements,
-              const UseMvcc use_mvcc = UseMvcc::Yes);
-
-  SQLPipeline(const std::string& sql, const std::shared_ptr<Optimizer>& optimizer,
-              const PreparedStatementCache& prepared_statements, const UseMvcc use_mvcc = UseMvcc::Yes);
-
-  // Explicit transaction context constructors
-  SQLPipeline(const std::string& sql, std::shared_ptr<TransactionContext> transaction_context);
-
-  SQLPipeline(const std::string& sql, const std::shared_ptr<Optimizer>& optimizer,
-              std::shared_ptr<TransactionContext> transaction_context);
-
-  SQLPipeline(const std::string& sql, const PreparedStatementCache& prepared_statements,
-              std::shared_ptr<TransactionContext> transaction_context);
-
-  SQLPipeline(const std::string& sql, const std::shared_ptr<Optimizer>& optimizer,
-              const PreparedStatementCache& prepared_statements,
-              std::shared_ptr<TransactionContext> transaction_context);
```

Comment:
When I look at all these constructors, it seems absolutely clear, why you wanted something else.